docs/RFCS/20190731_cluster_name.md

```diff
@@ -0,0 +1,110 @@
+- Feature Name: cluster name
+- Status: draft
+- Start Date: 2019-07-31
+- Authors: knz, ben, marc
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: #16784 #15888
+
+# Summary
+
+New feature: a string value, called "cluster name", and checked for
+equality when a newly started node joins a cluster.
+
+This will prevent newly added nodes to join the wrong cluster when a
+user has multiple clusters running side by side.
+
+This would be configured using another command-line parameter `--join-name`
+alongside `--join`.
+
+It will increase operational simplicity and the overall goal to “make
+data easy”
+
+Other impact:
+
+- (should have) it would be checked for equality in the node
+  certificate properties. This would prevent a node from trusting
+  another node's certificate if not in the same cluster, even though
+  they may be signed by the same CA.
+
+- (nice to have) the value would be shown in the admin UI, so that two UI
+  screens for different clusters side-by-side can be disambiguated at
+  a glance.
+
+- (to be further considered) the value would be used as an additional
+  label annotation on exported prometheus metrics. This would make it
+  possible for Grafana and other monitoring solutions to more easily
+  connect to multiple CockroachDB clusters simultaneously.
+
+# Motivation
+
+The motivation for doing this has been detail in
+https://github.com/cockroachdb/cockroach/issues/16784 and
+https://github.com/cockroachdb/cockroach/issues/15888
+
+- avoid mistaken node joins
+- disambiguate UI screens
+- disambiguate clusters in prometheus metrics
+
+# Guide-level explanation
+
+The service manager in charge of starting the `cockroach` process
+would activate this feature by providing the `--join-name` parameter
+on the command line.
+
+When the parameter is supplied, the node will verify that the other
+nodes that it connects to (via `--join`) have the same name
+configured. If they detect a different name, the join would abort.
+
+If the parameter is not supplied, we get the current behavior - the
+join succeeds in any case.
+
+# Reference-level explanation
+
+- The parameter is parsed for `cockroach start` only.
+
+- It is stored in the `server.Config` object.
+
+- When setting up TLS, it is used to validate the node certificate.
```

Comment:
Used to verify client certs (both nodes verifying clients and nodes verifying other nodes). Nodes verifying node server certs might be doable, but probably isn't needed.